maas/resource_maas_instance.go

```diff
@@ -197,6 +198,41 @@ func resourceMAASInstance() *schema.Resource {
 				Computed:    true,
 				Description: "The deployed MAAS machine pool name.",
 			},
+			"release_params": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Description: "Parameters used to release the allocated machine when the resource is destroyed.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"comment": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "A comment to be added to the event log when the machine is released.",
+						},
+						"erase": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Erase the disk when releasing.",
+						},
+						"force": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Force the release of the machine. If the machine was deployed as a KVM host, all machines inside the host will be deleted. Use with caution.",
```

Comment:
also yeah of course slightly update if this won't be a KVM host, but you've covered this already above